retrieval/raw/raw_endpoint.go

```diff
@@ -114,13 +124,145 @@ func downloadToFile(c *Client, method, sourceName string, workDir *os.File, URL
 		return filename, errors.New("Unable to create raw metric file")
 	}
 	defer util.SafeClose(rawRespFile.Close, &rerr)
-
 	filename = rawRespFile.Name()
 
+	if _, ok := ParsableFileSet[sourceName]; c.parseMetricData && ok {
+		err = parseAndWriteData(sourceName, resp.Body, rawRespFile)
+		return filename, err
+	}
+
 	_, err = io.Copy(rawRespFile, resp.Body)
 	if err != nil {
 		return filename, errors.New("Error writing file: " + rawRespFile.Name())
 	}
 
 	return filename, rerr
 }
+
+// TODO: investigate streamed json reading / writing
+func parseAndWriteData(filename string, reader io.Reader, writer io.Writer) error {
+	var to = getType(filename)
+	out := reflect.New(reflect.TypeOf(to))
+	err := json.NewDecoder(reader).Decode(out.Interface())
+
+	if err != nil {
+		return err
+	}
+	to = sanitizeData(out.Elem().Interface())
+
+	data, err := json.Marshal(to)
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(writer, bytes.NewReader(data))
+	return err
+}
+
+func getType(filename string) interface{} {
+	var to interface{}
+	switch filename {
+	case Nodes:
+		to = NodeList{}
+	case Namespaces:
+		to = NamespaceList{}
+	case Pods:
+		to = PodList{}
+	case PersistentVolumes:
+		to = PersistentVolumeList{}
+	case PersistentVolumeClaims:
+		to = PersistentVolumeClaimList{}
+	case AgentMeasurement:
+		to = CldyAgent{}
+	case Services, ReplicationControllers:
+		to = LabelMapMatchedResourceList{}
+	case Deployments, ReplicaSets, Jobs, DaemonSets:
+		to = LabelSelectorMatchedResourceList{}
+	}
+	return to
+}
+
+func sanitizeData(to interface{}) interface{} {
+	switch to.(type) {
+	case LabelSelectorMatchedResourceList:
+		return sanitizeSelectorMatchedResourceList(to)
+	case PodList:
+		return sanitizePodList(to)
+	case LabelMapMatchedResourceList:
+		return sanitizeMapMatchedResourceList(to)
+	case NamespaceList:
+		return sanitizeNamespaceData(to)
+	}
+	return to
+}
```

Comment:
Ok, get it. That is not obviously.. Why do we use two separate way to handle the files. A), only unmarshalling the data we need (defined in our struct). B) manually set some fields to empty after unmarshal

Reply:
Ideally it would all be explicitly defined objects that live in this repo but as I mentioned below that is, in my opinion, and unmaintainable mountain of data/structs. These objects were copied from the denormalizer and define the baseline set of data that we will need further down the pipeline and removes a small amount of high level data. `spec` for example is a first level key that only contains 1 piece of data that we need (labelselectors) out of a large number of fields and is therefore easy to parse out.

The further steps taken below where certain fields are nil'd out were done to remove sensitive or unused data that is highly nested